providers/edgedns/edgeDnsProvider.go

```diff
@@ -0,0 +1,252 @@
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/pkg/printer"
+	"github.com/StackExchange/dnscontrol/v3/pkg/txtutil"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+	"strings"
+)
+
+var features = providers.DocumentationNotes{
+	// The default for unlisted capabilities is 'Cannot'.
+	// See providers/capabilities.go for the entire list of capabilties.
+	providers.CanUseAlias:            providers.Cannot(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseDS:               providers.Cannot(),
+	providers.CanUseDSForChildren:    providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseNAPTR:            providers.Can(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanAutoDNSSEC:          providers.Can(),
+	providers.CantUseNOPURGE:         providers.Cannot(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.DocDualHost:            providers.Can(),
+	providers.CanUseSOA:              providers.Cannot(),
+	providers.DocCreateDomains:       providers.Can(),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanUseAKAMAICDN:        providers.Can(),
+}
+
+type edgeDNSProvider struct {
+	contractID string
+	groupID    string
+}
+
+func init() {
+	fns := providers.DspFuncs{
+		Initializer:   newEdgeDNSDSP,
+		RecordAuditor: AuditRecords,
+	}
+	providers.RegisterDomainServiceProviderType("EDGEDNS", fns, features)
+	providers.RegisterCustomRecordType("AKAMAICDN", "EDGEDNS", "")
+}
+
+// DnsServiceProvider
+func newEdgeDNSDSP(config map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	clientSecret := config["client_secret"]
+	host := config["host"]
+	accessToken := config["access_token"]
+	clientToken := config["client_token"]
+	contractID := config["contract_id"]
+	groupID := config["group_id"]
+
+	if clientSecret == "" {
+		return nil, fmt.Errorf("creds.json: client_secret must not be empty")
+	}
+	if host == "" {
+		return nil, fmt.Errorf("creds.json: host must not be empty")
+	}
+	if accessToken == "" {
+		return nil, fmt.Errorf("creds.json: accessToken must not be empty")
+	}
+	if clientToken == "" {
+		return nil, fmt.Errorf("creds.json: clientToken must not be empty")
+	}
+	if contractID == "" {
+		return nil, fmt.Errorf("creds.json: contractID must not be empty")
+	}
+	if groupID == "" {
+		return nil, fmt.Errorf("creds.json: groupID must not be empty")
+	}
+
+	EDInitialize(clientSecret, host, accessToken, clientToken)
+
+	api := &edgeDNSProvider{
+		contractID: contractID,
+		groupID:    groupID,
+	}
+	return api, nil
+}
+
+// AuditRecords returns an error if any records are not supportable by this provider.
+func AuditRecords(records []*models.RecordConfig) error {
```

Comment:
Please move the function `AuditRecords to a file called `auditrecords.go`

Reply:
Done.